support importing from dependencies

hypno2000 opened the pull request:
Thanks for this awesome library!

We needed to import components from dependencies. Implementing it lead me to making substantial changes how the library is configured. There is no app env config anymore, instead the root and namespace is now configured in app web file similar to how phoenix views are configured.

This pull request is a request for feedback. What do you think about this approach and do you think this should be merged here or stay a fork?

hypno2000 committed:
ability to import from external dependencies

hypno2000 committed:
readme

hypno2000 committed:
generator fixes

hypno2000 committed:
deps update

hypno2000 committed:
usage pattern consistent to phoenix

hypno2000 committed:
Update README.md

hypno2000 committed:
Update README.md

hypno2000 committed:
Update README.md

hypno2000 committed:
Update README.md

hypno2000 committed:
generator task rename

hypno2000 committed:
Merge branch 'master' of github.com:coingaming/phoenix_components

hypno2000 committed:
generator fix

hypno2000 committed:
generator fix

hypno2000 committed:
generator

hypno2000 committed:
generator fix

hypno2000 committed:
generator fix

hypno2000 committed:
generator fix

hypno2000 committed:
generator fix

hypno2000 committed:
generator fix

hypno2000 committed:
generator fix

hypno2000 committed:
linting

hypno2000 committed:
linting

hypno2000 committed:
linting

hypno2000 committed:
linting

hypno2000 committed:
linting

san650 commented:
Hi @hypno2000, thanks for opening the PR!

I like what you're are proposing here as a v2.0 version, and I would like to merge this. I need to take some time to review the PR to understand all the changes done.

I don't have time right now to dive into it, but I'll try to review it over the weekend.

Thanks!

anderson-amorim committed:
Shortcut for map form and field_name attrs

Added the option to use a custom component, when in a form, in a similar way as of the Phoenix html inputs. 
This options will map the form object(first arg) to the @attr :form and the field_name(second arg) to @attr :field_name automatically, therefore making the code more clear as developers can avoid unnecessary mappings.
Example: 
* Before
= currency_select form: f, field: :currency_iso

* After
= currency_select f, :currency_iso

* Comparing to Phoenix HTML input:
= text_input f, :description

hypno2000 committed:
Merge pull request #1 from anderson-amorim/patch-1

Shortcut for map form and field_name attrs

hypno2000 committed:
upgraded packages

hypno2000 committed:
comma

machineloop commented:
Hi @hypno2000 and @san650, bumping this thread. I've come to rely on some of these features and have been using the  fork for a component library I would like to make available. Wondering if it would be possible to start working towards a 2.0 release? i'm happy to pitch in, thanks!

danieldocki commented on lib/phoenix_components/view.ex:
```diff
@@ -100,23 +102,33 @@ defmodule PhoenixComponents.View do
+  
+  def component(namespace, name, %Phoenix.HTML.Form{} = form, field) when is_atom(field) do
+    do_component(namespace, name, "", [form: form, field: field])
+  end
```
I think this function is missing...

```elixir
  def component(namespace, name, %Phoenix.HTML.Form{} = form, field, do: block) when is_atom(field) do
    do_component(namespace, name, block, form: form, field: field)
  end
```

for something like that

```
<%= component MyWeb.Components, "form/field", f, :name do %>
  <%= component MyWeb.Components, "form/text_input", f, :name %>
<% end %>
```

Commit:
Changed component function calls with module and fixed tests

Commit:
Merge pull request #2 from tsenturk/ft/liveview-compability

Changed component function calls for LiveView compatibility with module and fixed tests
